neuron_morphology/transforms/scale_correction/compute_scale_correction.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, Optional, List, Any
+
+import logging
+import copy as cp
+
+import numpy as np
+import pandas as pd
+
+from argschema.argschema_parser import ArgSchemaParser
+from neuron_morphology.transforms.scale_correction._schemas import (
+    InputParameters, OutputParameters)
+from neuron_morphology.morphology import Morphology
+from neuron_morphology.swc_io import morphology_from_swc
+import neuron_morphology.transforms.affine_transform as aff
+
+
+def get_scale_correction(morphology: Morphology,
+                         cell_depth: float,
+                         soma_marker_z: float,
+                         cut_thickness: Optional[float] = 350):
+    """
+        Estimate a scale factor correction from recorded cell depth
+
+        Parameters
+        ----------
+        morphology: Morphology object
+        cell_depth: recorded depth of the soma when it was sliced
+        soma_marker_z: soma marker z value from revised marker file
+        cut_thickness: thickness of the cut slice
+
+        Returns
+        -------
+        scale factor correction
+
+    """
+    z_values = [node['z'] for node in morphology.nodes()]
+    max_z_extent = np.max(z_values) - np.min(z_values)
+
+    soma = morphology.get_soma()
+    fixed_depth = np.abs(soma['z'] - soma_marker_z)
```

Comment:
This module relies on relationships between soma['z'], soma_marker_z, and cell_depth, but I don't think it is ever explained what these relationships are. This seems pretty important because the only way to know these relationships is via familiarity with allen institute reconstruction procedures.